### PR TITLE
Make Cursor implement iterable

### LINKED ISSUE #33: Cursor should implement __iter__
The Cursor class in `dbapi.py` doesn't implement `__iter__` today meaning that code like `csv_writer.writerows(cursor)` or anything that tries to loop over the Cursor will not work.

### PR DESCRIPTION
**Pull request by hashhar**
Fixes #33.

### CODE REVIEW

**Comment by hashhar** [comment]
Tests are pending.

Also, the current impl ends up delegating to `PrestoResult::__iter__` which means that it's somewhat lazily evaluated.

**Comment by hashhar** [comment]
Existing tests are failing because co-ordinator was unable to start within 180 seconds on Travis. Tests are working locally though.

File: integration_tests/test_dbapi.py [hunk 1]
```diff
@@ -68,6 +68,20 @@ def test_select_query_result_iteration(presto_connection):
...
+    cur0.execute("select custkey from tpch.sf1.customer LIMIT 10")
+    rows0 = []
+    for row in cur0:
+        rows0.append(row)
+
+    cur1 = presto_connection.cursor()
+    cur1.execute("select custkey from tpch.sf1.customer LIMIT 10")
+    rows1 = cur1.fetchall()
+
+    assert len(rows0) == len(rows1)
```
Comment (findepi) [inline integration_tests/test_dbapi.py:82]: - you need ORDER BY to get stable results
- you can read from `nation`, test will be faster (then no need for LIMIT)

Comment (hashhar) [reply integration_tests/test_dbapi.py:82]: Changed to use `nation` and also asserts on the actual rows to in addition to length now.

**Comment by findepi** [comment]
Merged, thanks!

**Comment by findepi** [comment]
Released in `0.301.0`

**Pull request merged**